### ISSUE TITLE
Rustfmt warnings get prepended to the formatted output

### ISSUE THREAD

**Issue by Marwes** [rust-lang/rust.vim#250]
* rust.vim version: https://github.com/rust-lang/rust.vim/commit/e2cf315fb89dfcc3050da1a885129b1baa7f4889

Steps to reproduce:

* Create a rustfmt.toml file so rustfmt will pick it up and insert an invalid key `aaa = 1`.

* Format a file

Expected vs. actual behavior:

The warnings that rustfmt outputs should not be inserted at the top of the file, either it should be ignored or displayed in quickfix.

The problem is that when rustfmt is shelled out to with `system`, stdout and stderr gets merged into the output. Couldn't find a way to tell `system` to split them up however.

**Comment by da-x** [comment]
Thanks for looking into this!

So far I thought that rustfmt in stdout mode, either emits stderr or stdout, not both. So in this case, stderr was emitted, and the exit status was 0. 

Unless @nrc does not think it's a bug, then I think can think of two possible fixes: 1) Find a cross-platform way to redirect stderr to a temp-file (i.e. '2> <filename>'), and to read from that file. 2) Use Vim 8's async channels - AFAIK they can separate stderr and stdout. If running on an older Vim we can fallback to systemlist().

**Comment by nrc** [comment]
> Unless @nrc does not think it's a bug

I don't think this is a bug - it sounds like the output is going to stdout and the warnings/errors are going to stderr, which is expected behaviour.

**Comment by Marwes** [comment]
@nrc I think the (possible) bug in this case is that rustfmt could return a non-zero exit code in case of warnings (I think it did previously?). I'd say that would be poorer behavior though.
